lib/parse/data/parseEmbedData.ts

```diff
@@ -0,0 +1,94 @@
+import { IAudioData, IEmbedData, IRssItem } from '@interfaces/data';
+import { IEmbedConfig } from '@interfaces/embed/IEmbedConfig';
+import Parser from 'rss-parser';
+import parseAudioData from './parseAudioData';
+
+const parseEmbedData = (
+  config: IEmbedConfig,
+  rssData?: Parser.Output<Parser.Item>
+): IEmbedData => {
+  const {
+    feedUrl,
+    subscribeUrl,
+    imageUrl: configBgImageUrl,
+    title: configTitle,
+    subtitle: configSubtitle,
+    audioUrl: configAudioUrl,
+    epImageUrl: configImageUrl,
+    episodeGuid: configEpisodeGuid,
+    showPlaylist,
+    playlistSeason,
+    playlistCategory
+  } = config;
+  const {
+    title: rssSubtitle,
+    link: rssShareUrl,
+    image: rssImage,
+    itunes,
+    items: rssItems
+  } = rssData || {};
+  const { url: rssImageUrl } = rssImage || {};
+  const { image: rssItunesImage } = itunes || {};
+  const audioItems =
+    rssItems &&
+    rssItems.map((i) => ({
+      ...parseAudioData(i as IRssItem),
+      // Use feed title as audio items' subtitle.
+      subtitle: rssSubtitle
+    }));
+  const audio: IAudioData = {
+    // Establish defaults from feed props.
+    ...((rssImageUrl || rssItunesImage) && {
```

Comment:
I know that this is more usual in typescript/redux-y projects, and there's test coverage on this but...

I do find these nested object-spread-with-conditions tough to read.  Mostly curious if there are other patterns to use for this sort of stuff ... even `if (condition) { object.prop = 'value' }` feels a bit clearer.  But yeah, won't work on consts.

Reply:
This is not typescript or redux pattern, but more of an es5 javascript pattern. This is used to reduce the amount of code needed to do this in the order it needs to occur.

Reply:
Ah, right, ES5!  Yeah, not really a request for change... just me griping a bit about language-specific idioms.  Elixir has a ton of these shorthands too, where anyone not well versed in the language couldn't follow.